src/Microsoft.AspNetCore.Hosting/WebHostBuilder.cs

```diff
@@ -208,7 +209,7 @@ IServiceProvider GetProviderFromFactory(IServiceCollection collection)
                 var provider = collection.BuildServiceProvider();
                 var factory = provider.GetService<IServiceProviderFactory<IServiceCollection>>();
 
-                if (factory != null)
+                if (factory != null && !(factory is DefaultServiceProviderFactory))
```

Comment:
We could but then it would be over engineered. This also isn't 100% non-breaking. There's this edge case if you derived from `DefaultServiceProviderFactory `. It seems rare enough that i'm not worried about though.